api-channel-issue-tracker/src/main/java/com/synopsys/integration/alert/api/channel/issue/search/IssueTrackerSearcher.java

```diff
@@ -113,8 +114,13 @@ protected IssueTrackerSearcher(ProjectMessageToIssueModelTransformer modelTransf
                 searchResultOperation = policyOperation.get();
             } else if (optionalVulnerabilityDetails.isPresent()) {
                 IssueVulnerabilityDetails issueVulnerabilityDetails = optionalVulnerabilityDetails.get();
-                // TODO when ExistingIssueDetails has information about issue-status, use that to make a better choice of ItemOperation here.
-                searchResultOperation = issueVulnerabilityDetails.areAllComponentVulnerabilitiesRemediated() ? ItemOperation.DELETE : ItemOperation.ADD;
+                if (issueVulnerabilityDetails.areAllComponentVulnerabilitiesRemediated() && (IssueStatus.RESOLVABLE.equals(existingIssue.getIssueStatus()) || IssueStatus.UNKNOWN.equals(existingIssue.getIssueStatus()))) {
+                    searchResultOperation = ItemOperation.DELETE;
+                } else if (!issueVulnerabilityDetails.areAllComponentVulnerabilitiesRemediated() && (IssueStatus.REOPENABLE.equals(existingIssue.getIssueStatus()) || IssueStatus.UNKNOWN.equals(existingIssue.getIssueStatus()))) {
```

Comment:
The `if` and `else if` statements are a bit complex. Let's create variables for the sections after the `&&`.